### PR TITLE
translation(gh): translate smac、d4rl、icm

### PR DESCRIPTION
**Pull request by Khaiiiiiiiii**

### CODE REVIEW

File: d4rl.rst [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+    pip install gym
+    pip install -U 'mujoco-py<2.1'
+
+
+Verify installation
+--------
+
+After the installation is complete, you can verify that the installation was successful by running the following command on the Python command line:
+
+.. code::python
```
Comment (PaParaZz1) [inline d4rl.rst:116]: 少了个空格，语法未生效，`.. code:: python`

File: source/env_tutorial/d4rl.rst [hunk 2]
```diff
@@ -0,0 +1,358 @@
+D4RL (Mujoco)
```
Comment (PaParaZz1) [inline source/env_tutorial/d4rl.rst:1]: 重复了，把重复的删掉，路径为`source/env_tutorial/d4rl.rst`

File: source/env_tutorial/smac.rst [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+   cd DI-star
+   pip install -e .
+
+Verify installation
+--------
+
+After the installation is complete, you can confirm that the environment variable is set successfully by ``echo $SC2PATH`` after the installation is successful
+
+
+mirror
```
Comment (PaParaZz1) [inline source/env_tutorial/smac.rst:42]: 镜像这个词用Image，并且标题首字母大写

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: source/env_tutorial/smac.rst [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+Observation space
+------------------
+
+- You can obtain fragmentary information such as whether each agent is alive, the remaining HP of each agent, allies or enemies within the vision range of each agent.
+
+
+Action space
+---------------
+
+- The game operation button space, generally a discrete action space of size N (N varies with the specific sub-environment), the data type is \ ``int``\ , you need to pass in python values ​​(or 0-dimensional np arrays, such as actions 3 is \``np.array(3)``\)
```
Comment (VaninaY) [inline source/env_tutorial/smac.rst:62]: - The game operation button space, generally a discrete action space of size N (N varies with the specific sub-environment), the data type is \ ``int``\ , you need to pass in python values ​​(or 0-dimensional np arrays, such as actions 3 is\ ``np.array(3)``\ )

File: source/env_tutorial/smac.rst [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+
+   - 5: MOVE_WEST
+
+   - 6 - N: ATTACK ENEMY, the ID of the attacked enemy is N-6
+
+
+Bonus space
+--------
+
+- The game wins or loses, the victory is 1, and the defeat is 0, which is generally an \``int``\ value.
```
Comment (VaninaY) [inline source/env_tutorial/smac.rst:84]: - The game wins or loses, the victory is 1, and the defeat is 0, which is generally an \ ``int``\  value.

File: source/hands_on/ICM.rst [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+
+Key Equations or Key Graphs
+---------------------------------------
+The overall training and calculation process of the ICM algorithm is as follows:
+
+.. image:: images/ICM_illustraion.png
+   :align: center
+   :scale: 70%
+
+1. As shown in the figure on the left, the agent samples the action a in the state :math:`s_t` through the current policy :math:`\pi` and executes it, and finally obtains the state :math:`s_{t+1}`. The total reward is the sum of two partial rewards, one part is the external reward: math:`r_t^e`, that is, the sparse reward obtained in the environment; the other part is the intrinsic reward obtained by the ICM :math:`r_t^i` (The specific calculation process is given in step 4), the final strategy needs to achieve the purpose of training by optimizing the total reward.
```
Comment (VaninaY) [inline source/hands_on/ICM.rst:57]: 1. As shown in the figure on the left, the agent samples the action a in the state :math:`s_t` through the current policy :math:`\pi` and executes it, and finally obtains the state :math:`s_{t+1}`. The total reward is the sum of two partial rewards, one part is the external reward :math:`r_t^e`, that is, the sparse reward obtained in the environment; the other part is the intrinsic reward obtained by the ICM :math:`r_t^i` (The specific calculation process is given in step 4), the final strategy needs to achieve the purpose of training by optimizing the total reward.

File: source/hands_on/ICM.rst [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+2. In the reward module of ICM, it first extracts the eigenvectors after characterization by :math:`s_t` and :math:`s_{t+1}`,:math:`Phi(s_t; \theta_E)` and :math:`\Phi(s_{t+1}; \theta_E)` as input (reduce them to :math:`\Phi(s_t)` and :math:`\Phi(s_{t+1 })`), and output the predicted action value :math:`a_t`
+
+ :math:`\hat{a_t}=g(\Phi(s_t),\Phi(s_{t+1}) ; \theta_I)`
+
+ :math:`{\min}_{\theta_I, \theta_E} L_i(\hat{a_t},a_t)`
+
+Here :math:`\hat{a_t}` is the predicted value of :math:`a_t` and :math:`L_I` describes the difference between the two (cross entropy loss). The smaller the difference, the more accurate the estimation of the current action, and the better the representation of the environmental elements that the agent can control.
+
+
+3. The forward model of ICM will take :math:`Phi(s_t)`  and action value :math:`a_t` as input, and output the predicted value of the feature vector of the state at the next moment :math:`hat{\Phi} (s_{t+1})`
```
Comment (VaninaY) [inline source/hands_on/ICM.rst:73]: 3. The forward model of ICM will take :math:`\Phi(s_t)` and action value :math:`a_t` as input, and output the predicted value of the feature vector of the state at the next moment :math:`\hat{\Phi}(s_{t+1})` .

File: source/hands_on/ICM.rst [hunk 8]
```diff
@@ -0,0 +1,265 @@
...
+The error between the predicted feature vector at the next moment and the real feature vector is used as the intrinsic reward.
+
+ :math:`\hat{\phi(s_{t+1})}=f(\Phi(s_t),a_t) ; \theta_F)`
+
+ :math:`{\min}_{\theta_F, \theta_E} L_F(\hat{\phi(s_{t+1})},\phi(s_{t+1}))`
```
Comment (VaninaY) [inline source/hands_on/ICM.rst:78]: :math:`{\min}_{\theta_F, \theta_E} L_F(\hat{\phi(s_{t+1})},\phi(s_{t+1}))`

File: source/hands_on/ICM.rst [hunk 9]
```diff
@@ -0,0 +1,265 @@
...
+1. Reward normalization. Since the agent is in different stages and environments, the magnitude of the reward may change drastically. If it is directly used for subsequent calculations, it is easy to cause instability in subsequent learning.
+In our implementation, it is normalized to [0, 1] according to the following maximum and minimum normalization formula:
+
+``reward = (reward - reward.min()) / (reward.max() - reward.min() + 1e-8)``
+
+2. Use a residual network to fit the forward model. Since the representation dimension of observation is relatively large, the action value is often a discrete value.
+Therefore, when calculating the forward model, the residual network can better retain the information of the action value, so as to obtain a better environmental representation.
+
+
+.. code-block::python
```
Comment (VaninaY) [inline source/hands_on/ICM.rst:107]: .. code-block:: python

File: source/hands_on/ICM.rst [hunk 10]
```diff
@@ -0,0 +1,265 @@
...
+
+self.feature: extract the features of observation;
+
+self.inverse_net: The inverse model of the ICM network, which outputs a predicted action by taking two successive frames of feature features as input
+
+self.residual: Participate in the forward model of the ICM network, and make the features more obvious by concat the output of the action and the intermediate layer for many times
+
+self.forward_net: Participate in the forward model of the ICM network, responsible for outputting the feature at the moment of :math:`s_{t+1}`
+
+        .. code-block::python
```
Comment (VaninaY) [inline source/hands_on/ICM.rst:139]: .. code-block:: python

File: d4rl.rst [hunk 11]
```diff
@@ -0,0 +1,347 @@
...
+    git clone https://github.com/rail-berkeley/d4rl.git
+    cd d4rl
+    pip install -e .
+
+
+
+
+Mujoco only needs the gym and mujoco-py two libraries, which can be installed by one-click pip or combined with DI-engine
+
+1. The mujoco-py library no longer requires an activation license (``mujoco-py>=2.1.0``), you can install free-mujoco-py via \ `pip <https://github.com/openai/ mujoco-py/pull/640>`__ install
```
Comment (VaninaY) [inline d4rl.rst:60]: 1. The mujoco-py library no longer requires an activation license (``mujoco-py>=2.1.0``), you can \ `pip install free-mujoco-py <https://github.com/openai/mujoco-py/pull/640>`__  install

File: d4rl.rst [hunk 12]
```diff
@@ -0,0 +1,347 @@
+D4RL (Mujoco)
```
Comment (PaParaZz1) [inline d4rl.rst:1]: modify the location of `d4rl.rst`

File: d4rl.rst [hunk 13]
```diff
@@ -0,0 +1,347 @@
...
+
+- expert: train a \ `SAC <https://arxiv.org//abs/1801.01290>`__\ algorithm online until the strategy reaches the expert performance level, using the expert strategy to collect 1 million sample data
+- medium-expert: mixes equal amounts of data collected by expert and medium strategies
+- medium: first train a SAC algorithm online, stop training in the middle, and then use this partially trained strategy to collect 1 million sample data
+- medium-replay: trains a SAC algorithm online until the policy reaches a medium performance level, collecting all samples placed in the buffer during training
+- random: use a random initialization strategy to collect
+
+The picture below shows one of the Hopper games.
+
+.. image :: ./images/d4rl.gif
```
Comment (PaParaZz1) [inline d4rl.rst:35]: polish image

File: d4rl.rst [hunk 14]
```diff
@@ -0,0 +1,347 @@
...
+    tar xvf mujoco200-macos-x86_64.tar.gz && mkdir -p ~/.mujoco && mv mujoco200_macos ~/.mujoco/mujoco200
+    # Download unlocked activation key.
+    wget https://www.roboti.us/file/mjkey.txt -O ~/.mujoco/mjkey.txt
+    # Install and use mujoco-py
+    pip install gym
+    pip install -U 'mujoco-py<2.1'
+
+
+Verify Installation
+--------
```
Comment (PaParaZz1) [inline d4rl.rst:112]: it should be longer than head

File: d4rl.rst [hunk 15]
```diff
@@ -0,0 +1,347 @@
...
+   obs = env.reset()
+
+   while True:
+       action = env.random_action()
+       timestep = env.step(action)
+       if timestep.done:
+           print('Episode is over, final eval reward is: {}'.format(timestep.info['final_eval_reward']))
+           break
+
+DI-zoo Rnnable Code Example
```
Comment (PaParaZz1) [inline d4rl.rst:256]: typo

File: source/env_tutorial/smac.rst [hunk 16]
```diff
@@ -0,0 +1,368 @@
...
+   cd DI-star
+   pip install -e .
+
+Verify installation
+--------------------
+
+After the installation is complete, you can confirm that the environment variable is set successfully by ``echo $SC2PATH`` after the installation is successful
+
+
+Mirror
```
Comment (PaParaZz1) [inline source/env_tutorial/smac.rst:42]: use `Image` rather than `Mirror`

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

**Comment by Khaiiiiiiiii** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by Khaiiiiiiiii** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Pull request closed without merging**